modules/database/sqlite/wp-includes/sqlite/db.php

```diff
@@ -0,0 +1,256 @@
+			if ( is_array( $rewritten_query ) ) {
+				$table_query   = array_shift( $rewritten_query );
+				$index_queries = $rewritten_query;
+				$table_query   = trim( $table_query );
+				$pdo->exec( $table_query );
+			} else {
+				$rewritten_query = trim( $rewritten_query );
+				$pdo->exec( $rewritten_query );
+			}
+		}
+		$pdo->commit();
+		if ( $index_queries ) {
+			// $query_parser rewrites KEY to INDEX, so we don't need KEY pattern
+			$pattern = '/CREATE\\s*(UNIQUE\\s*INDEX|INDEX)\\s*IF\\s*NOT\\s*EXISTS\\s*(\\w+)?\\s*.*/im';
+			$pdo->beginTransaction();
+			foreach ( $index_queries as $index_query ) {
+				preg_match( $pattern, $index_query, $match );
+				$index_name = trim( $match[2] );
+				if ( in_array( $index_name, $index_array, true ) ) {
+					$r           = rand( 0, 50 );
+					$replacement = $index_name . "_$r";
+					$index_query = str_ireplace(
+						'EXISTS ' . $index_name,
+						'EXISTS ' . $replacement,
+						$index_query
+					);
+				} else {
+					$index_array[] = $index_name;
+				}
+				$pdo->exec( $index_query );
+			}
+			$pdo->commit();
+		}
+	} catch ( PDOException $err ) {
+		$err_data = $err->errorInfo; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$err_code = $err_data[1];
+		if ( 5 == $err_code || 6 == $err_code ) { // phpcs:ignore WordPress.PHP.StrictComparisons.LooseComparison
+			// If the database is locked, commit again.
+			$pdo->commit();
+		} else {
+			$pdo->rollBack();
+			$message  = sprintf(
+				'Error occurred while creating tables or indexes...<br />Query was: %s<br />',
+				var_export( $rewritten_query, true )
+			);
+			$message .= sprintf( 'Error message is: %s', $err_data[2] );
+			wp_die( $message, 'Database Error!' );
+		}
+	}
+
+	$query_parser = null;
+	$pdo          = null;
+
+	return true;
+}
+
+/**
+ * Installs the site.
+ *
+ * Runs the required functions to set up and populate the database,
+ * including primary admin user and initial options.
+ *
+ * @since n.e.x.t.
+ *
+ * @param string $blog_title    Site title.
+ * @param string $user_name     User's username.
+ * @param string $user_email    User's email.
+ * @param bool   $is_public     Whether the site is public.
+ * @param string $deprecated    Optional. Not used.
+ * @param string $user_password Optional. User's chosen password. Default empty (random password).
+ * @param string $language      Optional. Language chosen. Default empty.
+ * @return array {
+ *     Data for the newly installed site.
+ *
+ *     @type string $url              The URL of the site.
+ *     @type int    $user_id          The ID of the site owner.
+ *     @type string $password         The password of the site owner, if their user account didn't already exist.
+ *     @type string $password_message The explanatory message regarding the password.
+ * }
+ */
+function wp_install( $blog_title, $user_name, $user_email, $is_public, $deprecated = '', $user_password = '', $language = '' ) {
```

Comment:
Is this polyfill version of WordPress function? Do we needs to check `if ( ! function_exists( 'wp_install' ) ) {`?

Reply:
`Is this polyfill version of WordPress function?`

Yes. When using SQLite there are a few things that need to be different; Specifically, we replace the call to `make_db_current_silent()` with `perflab_sqlite_make_db_sqlite()`. Other than that, the function is identical with Core.

> Do we needs to check `if ( ! function_exists( 'wp_install' ) ) {`?
We don't need a `function_exists` check, this is loaded properly on time, and overrides the Core `wp_install` function - which is wrapped in a `function_exists` check 👍

Reply:
Make sense.